src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -265,22 +265,12 @@ protected function prepare_item_for_database( $request ) {
 				$config['styles'] = $existing_config['styles'];
 			}
 
-			/*
-			 * If the incoming request is going to create a new variation
-			 * that is not yet registered, we register it here.
-			 * This is because the variations are registered on init,
-			 * but we want this endpoint to return the new variation immediately:
-			 * if we don't register it, it'll be stripped out of the response
-			 * just in this request (subsequent ones will be ok).
-			 * Take the variations defined in styles.blocks.variations from the incoming request
-			 * that are not part of the $exsting_config.
-			 */
+			// Register theme-defined variations.
+			WP_Theme_JSON_Resolver::get_theme_data();
+
+			// Register user-defined variations.
 			if ( isset( $request['styles']['blocks']['variations'] ) ) {
-				$existing_variations = isset( $existing_config['styles']['blocks']['variations'] ) ? $existing_config['styles']['blocks']['variations'] : array();
-				$new_variations      = array_diff_key( $request['styles']['blocks']['variations'], $existing_variations );
-				if ( ! empty( $new_variations ) ) {
-					wp_register_block_style_variations_from_theme_json_data( $new_variations );
-				}
+				wp_register_block_style_variations_from_theme_json_data( $config['styles']['blocks']['variations'] ?? array() );
```

Comment:
Could we do an empty check on `$config['styles']['blocks']['variations']`? 

I see `wp_register_block_style_variations_from_theme_json_data` does for the incoming argument anyway, but it would save having to pass an empty array. 

As it is, I'm not sure if passing `array()` is meant to do anything or not.

Reply:
Addressed at https://github.com/WordPress/wordpress-develop/pull/6844/commits/d6fda7e5be309e49605cc44d566aeafe4ee2fad3